sql/core/src/main/scala/org/apache/spark/sql/execution/joins/CartesianProduct.scala

```diff
@@ -28,9 +28,17 @@ import org.apache.spark.sql.execution.metric.SQLMetrics
  * :: DeveloperApi ::
  */
 @DeveloperApi
-case class CartesianProduct(left: SparkPlan, right: SparkPlan) extends BinaryNode {
+case class CartesianProduct(
+    left: SparkPlan,
+    right: SparkPlan,
+    buildSide: BuildSide) extends BinaryNode {
```

Comment:
@yhuai use buildSide just want to know which side is small, and use this to decide whether we need to change the order.